src/Data/Map/Internal.purs

```diff
@@ -158,6 +155,52 @@ instance traversableWithIndexMap :: TraversableWithIndex k (Map k) where
           <*> f k2 v2
           <*> traverseWithIndex f right
 
+-- | `SemigroupMap k v` provides a `Semigroup` instance for `Map` whose definition
+-- | depends on the underlying `value` type's `Semigroup` instance.
+-- | You should only use this type when you need `Data.Map` to have
+-- | a `Semigroup` instance.
+-- |
+-- | ```purescript
+-- | let
+-- |   s :: forall key value. key -> value -> SemigroupMap key value
+-- |   s k v = Unbiased.SemigroupMap (Data.Map.singleton k v)
+-- |
+-- | (s 1     "foo") <> (s 1     "bar") == (s 1  "foobar")
+-- | (s 1 (First 1)) <> (s 1 (First 2)) == (s 1 (First 1))
+-- | (s 1  (Last 1)) <> (s 1  (Last 2)) == (s 1  (Last 1))
+-- | ```
+newtype SemigroupMap k v = SemigroupMap (Map k v)
+
+type role SemigroupMap nominal representational
+
+derive newtype instance eq1SemigroupMap :: Eq k => Eq1 (SemigroupMap k)
+derive newtype instance eqSemigroupMap :: (Eq k, Eq v) => Eq (SemigroupMap k v)
+derive newtype instance ord1SemigroupMap :: Ord k => Ord1 (SemigroupMap k)
+derive newtype instance ordSemigroupMap :: (Ord k, Ord v) => Ord (SemigroupMap k v)
+derive instance newtypeSemigroupMap :: Newtype (SemigroupMap k v) _
+
+instance showSemigroupMap :: (Show k, Show v) => Show (SemigroupMap k v) where
+  show (SemigroupMap m) = "(fromFoldable " <> show (toAscArray m) <> ")"
```

Comment:
Can’t this instance be newtype derived?

Reply:
Nope:
```
  Cannot derive a type class instance for
  
    Data.Show.Show (SemigroupMap k v)
  
  since instances of this type class are not derivable.
```

Reply:
I meant to _newtype_ derive the instance: 

```purs
derive newtype instance showSemigroupMap :: (Show k, Show v) => Show (SemigroupMap k v)
```

Reply:
Wow.... Duh. Sorry about that. Fixed.